app/components/FileRow.js

```diff
@@ -216,7 +217,7 @@ export default class FileRow extends Component {
 
     const metadata = file.game.getMetadata() || {};
     const startAt = metadata.startAt;
-    const startAtDisplay = timeUtils.convertToDateAndTime(startAt) || 'Unknown';
+    const startAtDisplay = timeUtils.convertToDateAndTime(startAt) || timeUtils.filenameToDateAndTime(this.getFileName()) || timeUtils.convertToDateAndTime(fs.statSync(file.fullPath).birthtime) || 'Unknown';
```

Comment:
Separate each function call into its own variable with a meaningful name

Reply:
Not sure if that's necessarily a good idea in this case as it will cause them all to actually be evaluated. Probably best to avoid calculating esp the creation time one if it's not needed.

Reply:
One option to get around this is to make variables like:
`const getFromFileName = () => timeUtils.filenameToDateAndTime(this.getFileName())`

Reply:
Unfortunately doesn't work as functions can't be defined within React child functions - thinking of another solution for this though

Reply:
Honestly I think if you can make a function in timeUtils itself that goes down the list of priority without executing the ones that aren't necessary would be ideal.

Reply:
In order to get sorting right you may need to calculate the start time at the file level instead so it can be used for sorting properly.